File: 0392-is-subsequence/0392-is-subsequence.cpp
// Time Complexity is O(N) where n is the size of the target string.
// Space Complexity is O(1)
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int i=0 , j=0;
        
        while(i<s.size() && j<t.size()){
            if(s[i] == t[j]){
                i++;
            }
            j++;
        }
        if(i==s.size()){
            return true;
        }
        return false;
        
    }
};

/*
int main()
{
Solution soluiton;
string s = "abc";
string t = "ahbgdc";

bool result = solution.isSubsequence(s,t);

cout << "Is s a subsequence of t? " << (result ? "Yes" : "No") << endl;

return 0;
}
*/

File: 0392-is-subsequence/0392-is-subsequence.py
##Time Complexity is O(N) where n is the size of the target string.
##Space Complexity is O(1)
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        i,j = 0,0
        
        while(i<len(s) and j<len(t)):
            if(s[i] == t[j]):
                i+=1
            j+=1
        
        return i == len(s)
    
#Solution = solution()
# s ="abc"
#t = "ahbgdc"

#result = solution.isSubsequence(s,t)

#print(result)
        